### DISCUSSION TITLE
Wrong type when using withImmer and atomWithStorage

### DISCUSSION THREAD

**Discussion by OsamaAburabie** [pmndrs/jotai#1492] [Q&A]
I'm trying to create an atomWithStorage using withImmer.
I'm using AsyncStorage in React Native.

when I pass the content type it throws an error in the storage object type.

<img width="635" alt="image" src="https://user-images.githubusercontent.com/67992414/193264116-eb9bd8af-9788-4c1a-bb34-680472851999.png">

<img width="539" alt="image" src="https://user-images.githubusercontent.com/67992414/193264159-998a5124-e619-4030-a221-4dfe66531f5e.png">

I'm I doing something wrong?

**Comment by dai-shi** [comment]
The first issue can be reproduced without immer, right?
It seems like you need to annotate `storage` type.
